src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java

```diff
@@ -84,13 +84,20 @@ public ReplicateSupplyService(ReplicatesService replicatesService,
         }
 
         for (Task task : runningTasks) {
+            String chainTaskId = task.getChainTaskId();
+
+            taskService.initializeTaskAccessForNewReplicate(chainTaskId);
+            if (taskService.isTaskAccessedForNewReplicate(chainTaskId)){
+                continue;//skip task if being accessed
+            }
+            taskService.setTaskAccessedForNewReplicate(chainTaskId, true);//lock task while being accessed
```

Comment:
We better call the method XXXLock explicitly instead of setXXX.

Reply:
Right!